<!doctype html> <html lang=en > <meta charset=UTF-8 > <meta name=viewport  content="width=device-width, initial-scale=1"> <link rel=stylesheet  href="/css/franklin.css"> <link rel=stylesheet  href="/css/basic.css"> <link rel=icon  href="/assets/favicon.png"> <title>Single-cell data analysis with Scanpy and scvi-tools</title> <header> <div class=blog-name ><a href="">CompBio Skills Seminar</a></div> <nav> <ul> <li><a href="/">Home</a> <li><a href="/resources/">Resources</a> </ul> <img src="/assets/hamburger.svg" id=menu-icon > </nav> </header> <div class=franklin-content ><h1 id=single-cell_data_analysis_with_scanpy_and_scvi-tools ><a href="#single-cell_data_analysis_with_scanpy_and_scvi-tools" class=header-anchor >Single-cell data analysis with Scanpy and scvi-tools</a></h1> <p>A flourishing body of computational tools have made it easier to robustly analyze single-cell -omics datasets in a scalable and reproducible way. Here we will dive into conducting an analysis of a single-cell RNA-sequencing dataset with <a href="https://scanpy.readthedocs.io/en/stable/">Scanpy</a> and <a href="https://scvi-tools.org">scvi-tools</a>, two popular Python libraries for general purpose analysis tasks.</p> <p>This tutorial will cover the following items:</p> <ol> <li><p>Overview of the <a href="https://anndata.readthedocs.io/en/latest/">AnnData</a> format, which powers Python-based single-cell libraries</p> <li><p>Data preprocessing and quality control</p> <li><p>Dimensionality reduction and dataset integration</p> <li><p>Differential expression</p> <li><p>Visualization</p> </ol> <p>There will be a focus not only on how to run the software, but also the motivation behind each of these steps.</p> <h2 id=notebook ><a href="#notebook" class=header-anchor >Notebook</a></h2> <p>The colab notebook can be found <a href="https://colab.research.google.com/drive/1V4BD3SAGDwLzvMUn90FMOHYVNG_iP4Ee?usp&#61;sharing">here</a>. The full notebook can also be viewed in a web browser <a href="/assets/scvi_notebook.html">here</a>.</p> <h2 id=relevant_references ><a href="#relevant_references" class=header-anchor >Relevant references</a></h2> <ul> <li><p>Andrews, T. S., &amp; Hemberg, M. &#40;2018&#41;. False signals induced by single-cell imputation. F1000Research, 7.</p> <li><p>Lopez, R., Regier, J., Cole, M. B., Jordan, M. I., &amp; Yosef, N. &#40;2018&#41;. Deep generative modeling for single-cell transcriptomics. Nature methods, 15&#40;12&#41;, 1053-1058.</p> <li><p>Luecken, M. D., &amp; Theis, F. J. &#40;2019&#41;. Current best practices in single‐cell RNA‐seq analysis: a tutorial. Molecular systems biology, 15&#40;6&#41;, e8746.</p> <li><p>Wolf, F. A., Angerer, P., &amp; Theis, F. J. &#40;2018&#41;. SCANPY: large-scale single-cell gene expression data analysis. Genome biology, 19&#40;1&#41;, 15.</p> </ul> <h2 id=other_resources ><a href="#other_resources" class=header-anchor >Other resources</a></h2> <ul> <li><p><a href="https://theislab.github.io/scanpy-in-R/">Scanpy in R</a></p> <li><p><a href="https://www.scvi-tools.org/en/stable/user_guide/notebooks/scvi_in_R.html">scvi-tools in R</a></p> <li><p>Conversion of data objects with <a href="https://bioconductor.org/packages/release/bioc/html/zellkonverter.html">Zellkonverter</a></p> </ul> <div class=page-foot > <div class=copyright > &copy; Adam Gayoso, Matt Jones. Last modified: March 09, 2022. Website built with <a href="https://github.com/tlienart/Franklin.jl">Franklin.jl</a>. </div> </div> </div>